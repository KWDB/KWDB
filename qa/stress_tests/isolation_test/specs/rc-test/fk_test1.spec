setup
{
  CREATE TABLE foo (a int PRIMARY KEY, b text);
  CREATE TABLE bar (a int NOT NULL REFERENCES foo);
  INSERT INTO foo VALUES (42);
}

teardown
{
  DROP TABLE foo, bar;
}

session s1
step s1		{ BEGIN transaction isolation level read committed; }
step s1w	{ INSERT INTO bar VALUES (42); }
step s1r1   { select * from foo; }
step s1r2   { select * from bar; }
step s1c	{ COMMIT; }

session s2
step s2     { BEGIN transaction isolation level read committed; }
step s2u    { UPDATE foo SET b = 'Hello World'; }
step s2r1   { select * from foo; }
step s2r2   { select * from bar; }
step s2c    { COMMIT; }

permutation s1 s2 s1w s2u s1r1 s1r2 s1c s2r1 s2r2 s2c
permutation s1 s2 s1w s2u s2r1 s2r2 s2c s1r1 s1r2 s1c
permutation s1 s2 s1w s2r1 s2r2 s2u s1r1 s1r2 s1c s2c
permutation s1 s2 s1w s2r1 s2r2 s2u s1r1 s1r2 s2c s1c
