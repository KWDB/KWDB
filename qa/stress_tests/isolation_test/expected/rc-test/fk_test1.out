Parsed test spec with 2 sessions

starting permutation: s1 s2 s1w s2u s1r1 s1r2 s1c s2r1 s2r2 s2c
step s1: BEGIN transaction isolation level read committed;  finished.
step s2: BEGIN transaction isolation level read committed;  finished.
step s1w: INSERT INTO bar VALUES (42);  finished.
step s2u: UPDATE foo SET b = 'Hello World';  finished.
step s1r1: select * from foo;  finished.
 a|b
--+-
42| 
(1 row)

step s1r2: select * from bar;  finished.
 a
--
42
(1 row)

step s1c: COMMIT;  finished.
step s2r1: select * from foo;  finished.
 a|b          
--+-----------
42|Hello World
(1 row)

step s2r2: select * from bar;  finished.
 a
--
42
(1 row)

step s2c: COMMIT;  finished.

starting permutation: s1 s2 s1w s2u s2r1 s2r2 s2c s1r1 s1r2 s1c
step s1: BEGIN transaction isolation level read committed;  finished.
step s2: BEGIN transaction isolation level read committed;  finished.
step s1w: INSERT INTO bar VALUES (42);  finished.
step s2u: UPDATE foo SET b = 'Hello World';  finished.
step s2r1: select * from foo;  finished.
 a|b          
--+-----------
42|Hello World
(1 row)

step s2r2: select * from bar;  finished.
a
-
(0 rows)

step s2c: COMMIT;  finished.
step s1r1: select * from foo;  finished.
 a|b          
--+-----------
42|Hello World
(1 row)

step s1r2: select * from bar;  finished.
 a
--
42
(1 row)

step s1c: COMMIT;  finished.

starting permutation: s1 s2 s1w s2r1 s2r2 s2u s1r1 s1r2 s1c s2c
step s1: BEGIN transaction isolation level read committed;  finished.
step s2: BEGIN transaction isolation level read committed;  finished.
step s1w: INSERT INTO bar VALUES (42);  finished.
step s2r1: select * from foo;  finished.
 a|b
--+-
42| 
(1 row)

step s2r2: select * from bar;  finished.
a
-
(0 rows)

step s2u: UPDATE foo SET b = 'Hello World';  finished.
step s1r1: select * from foo;  finished.
 a|b
--+-
42| 
(1 row)

step s1r2: select * from bar;  finished.
 a
--
42
(1 row)

step s1c: COMMIT;  finished.
step s2c: COMMIT;  finished.

starting permutation: s1 s2 s1w s2r1 s2r2 s2u s1r1 s1r2 s2c s1c
step s1: BEGIN transaction isolation level read committed;  finished.
step s2: BEGIN transaction isolation level read committed;  finished.
step s1w: INSERT INTO bar VALUES (42);  finished.
step s2r1: select * from foo;  finished.
 a|b
--+-
42| 
(1 row)

step s2r2: select * from bar;  finished.
a
-
(0 rows)

step s2u: UPDATE foo SET b = 'Hello World';  finished.
step s1r1: select * from foo;  finished.
 a|b
--+-
42| 
(1 row)

step s1r2: select * from bar;  finished.
 a
--
42
(1 row)

step s2c: COMMIT;  finished.
step s1c: COMMIT;  finished.
